tutorials/sns-instances-notification-system/index.mdx

```diff
@@ -0,0 +1,292 @@
+ }
+```
+
+5. Create the publisher server instance's ip address:
+```
+ resource "scaleway_instance_ip" "publisher_public_ip" {
+   project_id = scaleway_account_project.sns_tutorial.id
+   zone = "fr-par-1"
+ }
+```
+
+6. Create the instance itself with all the environment variables needed to publish the notifications on the topic:
+```
+ resource "scaleway_instance_server" "publisher_sns_tuto_instance" {
+   project_id = scaleway_account_project.sns_tutorial.id
+   name       = "publisher-server"
+   type       = "PLAY2-PICO"
+   image      = "debian_bookworm"
+   ip_id      = scaleway_instance_ip.publisher_public_ip.id
+   security_group_id= scaleway_instance_security_group.sns_www.id
+
+   # USER DATA TO RUN THE SERVER AT START-UP WITH A SCRIPT TO SET THE SNS ENVIRONMENT VARIABLES RELATED TO THE TOPIC
+   user_data = {
+     cloud-init = <<-EOF
+     #cloud-config
+     write_files:
+       - content: |
+           #!/bin/bash
+           export TOPIC_ARN="${scaleway_mnq_sns_topic.topic.arn}"
+           export AWS_ACCESS_KEY="${scaleway_mnq_sns_credentials.main.access_key}"
+           export AWS_SECRET_KEY="${scaleway_mnq_sns_credentials.main.secret_key}"
+         owner: root:root
+         path: /etc/profile.d/publisher-server_env.sh
+         permissions: '0755'
+     runcmd:
+       - apt-get update && apt-get install -y docker.io
+       - systemctl start docker
+       - systemctl enable docker
+       - chmod +x /etc/profile.d/publisher-server_env.sh
+       # Using 'source' command doesn't work in cloud-init runcmd because each command runs in a separate shell.
+       # this is why environment variables will be passed directly in the docker run command
+       - docker pull rg.fr-par.scw.cloud/sns-x-instance-tutorial/publisher-server:1.0
+       - |
+         . /etc/profile.d/publisher-server_env.sh
+         docker run -d --restart=always \
+           --name publisher-server \
+           -p 8081:8081 \
+           -e TOPIC_ARN=$TOPIC_ARN \
+           -e AWS_ACCESS_KEY=$AWS_ACCESS_KEY \
+           -e AWS_SECRET_KEY=$AWS_SECRET_KEY \
+           rg.fr-par.scw.cloud/sns-x-instance-tutorial/publisher-server:1.0
+   EOF
+   }
+ }
+```
+
+# Use the publisher and subscriber websites to test your setup
+
+1. Run the command `terraform apply`. You'll be prompt to enter your public SSH key. Paste the result of `cat ~/.ssh/id_ed25519.pub` and enter `yes`
+2. You need to confirm the subscription to the topic on the subscriber server. Go on the console, Instance, and copy the `subscriber_server` IP address.
+3. Then go on the home page of your subscriber server: ```http://<your_subscriber_ip_address>:8081```
+4. Click on `Confirm subscription`. If you have an error because the URL hasn't been received, you can reload the page. It should take less than 10 seconds to appear.
+5. Once you've arrived to the AWS subscription confirmation page, go back to the home page, and click on `Notifications`
```

Comment:
This step is a little confusing to me, by home page do we mean the home page of the subscriber Instance?

Reply:
on the index.html reached with this url : ```http://<your_subscriber_ip_address>:8081```

Reply:
I think we shouldn't mention the AWS subscription confirmation page in this step then, if I've understood it is the page they are directed to in the previous step?

Maybe we can have:

4. Still in the browser, click `Confirm subscription` to confirm the subscription to the SNS topic we created. You are directed to an AWS page which confirms the confirmation of your subscription. If you have an error because the URL hasn't been received, you can reload the page. It should take less than 10 seconds to appear.

5. Go back to the Instance subscriber's home page at http://<your_subscriber_ip_address>:8081 and click `Notifications`. Keep this page open while you carry out the next steps in a new browser tab.